adam-core/src/main/scala/org/bdgenomics/adam/rdd/BroadcastRegionJoin.scala

```diff
@@ -165,7 +165,7 @@ object BroadcastRegionJoin extends RegionJoin {
  *
  * @param regions The input-set of regions.
  */
-private[rdd] class NonoverlappingRegions(regions: Iterable[ReferenceRegion]) extends Serializable {
```

Comment:
Why are we making these public? IMO, they should stay package private or move to the `org.bdgenomics.adam.models` package.